app/components/support/crate-report-form.js

```diff
@@ -0,0 +1,135 @@
+    description: 'it contains a vulnerability (please try to contact the crate author first)',
+  },
+  {
+    reason: 'other',
+    description: 'it is violating the usage policy in some other way (please specify below)',
+  },
+];
+
+export default class CrateReportForm extends Component {
+  @service store;
+  @alias('loadCrateTask.last.value') crate;
+
+  @tracked name = '';
+  @tracked selectedReasons = [];
+  @tracked detail = '';
+  @tracked nameInvalid;
+  @tracked reasonsInvalid;
+  @tracked detailInvalid;
+
+  reasons = REASONS;
+
+  constructor() {
+    super(...arguments);
+    this.reset();
+    this.name = this.args.name;
+  }
+
+  reset() {
+    this.nameInvalid = false;
+    this.reasonsInvalid = false;
+    this.detailInvalid = false;
+  }
+
+  validate() {
+    this.nameInvalid = !this.crate;
+    this.reasonsInvalid = this.selectedReasons.length === 0;
+    this.detailInvalid = this.selectedReasons.includes('other') && !this.detail?.trim();
+    return !this.nameInvalid && !this.reasonsInvalid && !this.detailInvalid;
+  }
+
+  @action resetNameValidation() {
+    this.nameInvalid = false;
+  }
+
+  @action resetDetailValidation() {
+    this.detailInvalid = false;
+  }
+
+  @action isReasonSelected(reason) {
+    return this.selectedReasons.includes(reason);
+  }
+
+  @action toggleReason(reason) {
+    this.selectedReasons = this.selectedReasons.includes(reason)
+      ? this.selectedReasons.filter(it => it !== reason)
+      : [...this.selectedReasons, reason];
+    this.reasonsInvalid = false;
+  }
+
+  @action
+  async submit() {
+    // Verify name
+    await this.loadCrateTask.perform().catch(() => {
+      // ignore all errors
+    });
+    if (!this.validate()) {
+      return;
+    }
+
+    let mailto = this.composeMail();
+    window.open(mailto, '_self');
+  }
+
+  composeMail() {
+    let name = this.name;
+    let reasons = this.reasons
+      .map(({ reason, description }) => {
+        let selected = this.isReasonSelected(reason);
+        return `${selected ? '- [x]' : '- [ ]'} ${description}`;
+      })
+      .join('\n');
+    let body = `I'm reporting the https://crates.io/crates/${name} crate because:
+
+${reasons}
+
+Additional details:
+
+${this.detail}
+`;
+    let subject = `The "${name}" crate`;
+    let address = 'help@crates.io';
+    let mailto = `mailto:${address}?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`;
+    return mailto;
+  }
+
+  loadCrateTask = dropTask(async () => {
+    if (!this.name) return;
+    // Try reusing crate in the store if name is given
+    // and fallback to fetch if its not in store
+    if (!this.crate && this.args.name === this.name) {
+      let crate = this.store.peekRecord('crate', this.args.name);
+      if (crate) {
+        return crate;
+      }
+    }
+    let last = this.loadCrateTask.lastSuccessful?.value;
+    return last?.id === this.name ? last : await ajax(`/api/v1/crates/${this.name}?include=`).then(r => r.crate);
```

Comment:
`this.store.findRecord(...)` would be easier, or am I missing something?

Reply:
Yes, `findRecord` definitely does the work more simply. The reason for using ajax here is that `include=` is more efficient because it doesn't query for unnecessary information.

Reply:
we could add support for `?include=` to our Ember Data adapter though (see "Retrieving Related Model Records" in https://api.emberjs.com/ember-data/5.3/classes/Store/methods/findRecord?anchor=findRecord)

Reply:
No longer a concern, as we do not verify the existence of the crate anymore.